core/iwasm/libraries/debug-engine/gdbserver.c

```diff
@@ -119,57 +132,169 @@ wasm_close_gdbserver(WASMGDBServer *server)
 }
 
 static inline void
-handler_packet(WASMGDBServer *server, char request, char *payload)
+handle_packet(WASMGDBServer *server, char request, char *payload)
 {
     if (packet_handler_table[(int)request].handler != NULL)
         packet_handler_table[(int)request].handler(server, payload);
 }
 
+static void
+process_packet(WASMGDBServer *server)
+{
+    uint8 *inbuf = (uint8 *)server->receive_ctx->receive_buffer;
+    char request;
+    char *payload = NULL;
+
+    request = inbuf[0];
+    payload = (char *)&inbuf[1];
+
+    if (request == '\0') {
+        LOG_VERBOSE("ignore empty request");
+        return;
+    }
```

Comment:
Should move these lines before L149? If inbuf[0] is '\0', we should not read inbuf[1] in L149?